### PR TITLE
[Execution] adding edge case tests cadence program cache

### PR DESCRIPTION
**Pull request by ramtinms**

### CODE REVIEW

File: engine/execution/computation/programs_test.go [hunk 1]
```diff
@@ -0,0 +1,368 @@
...
+
+}
+
+func createTestBlockAndRun(t *testing.T, engine *Manager, parentBlock *flow.Block, col flow.Collection, view *delta.View) (*flow.Block, *execution.ComputationResult) {
+	guarantee := flow.CollectionGuarantee{
+		CollectionID: col.ID(),
+		Signature:    nil,
+	}
+
+	block := &flow.Block{
```
Comment (m4ksio) [inline engine/execution/computation/programs_test.go:326]: very minor, but we have fixtures to create blocks with parents.

File: engine/execution/computation/programs_test.go [hunk 2]
```diff
@@ -0,0 +1,368 @@
...
+	t.Run("executing block11 (deploys contract version 1)", func(t *testing.T) {
+		block11tx1 := testutil.DeployEventContractTransaction(account, chain, 1)
+		prepareTx(t, block11tx1, account, privKey, 0, chain)
+
+		txs11 := []*flow.TransactionBody{block11tx1}
+		col11 := flow.Collection{Transactions: txs11}
+		block11View = block1View.NewChild()
+		block11, res = createTestBlockAndRun(t, engine, block1, col11, block11View)
+		// cache should include value for this block
+		require.NotNil(t, programsCache.Get(block11.ID()))
```
Comment (m4ksio) [inline engine/execution/computation/programs_test.go:201]: Maybe its worth to add extra check with `HasChanges()`, just for sanity check. Essentially can be applied everywhere we asses programs

**Review by m4ksio — APPROVED** [review]
Great work Ramtin! Really good test cases

File: engine/execution/computation/programs_test.go (hunk 2 above)
Comment (ramtinms) [reply engine/execution/computation/programs_test.go:201]: added some cases

File: engine/execution/computation/programs_test.go (hunk 1 above)
Comment (ramtinms) [reply engine/execution/computation/programs_test.go:330]: I would like to keep it this way, atleast for now, I want to expand this test for some other fvm stuff as well and prefer not to update the original fixture.
